File: tools/bench/answers/go_haiku/s2/05_dedup.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	nums := []int{3, 1, 3, 2, 1, 4}
	seen := make(map[int]bool)
	var result []int
	for _, n := range nums {
		if !seen[n] {
			seen[n] = true
			result = append(result, n)
		}
	}
	var strNums []string
	for _, n := range result {
		strNums = append(strNums, fmt.Sprintf("%d", n))
	}
	fmt.Println(strings.Join(strNums, " "))
}

File: tools/bench/answers/rust_sonnet/s1/05_dedup.rs
fn main() {
    let nums = [3, 1, 3, 2, 1, 4];
    let mut seen = Vec::new();
    let mut result = Vec::new();
    for &n in &nums {
        if !seen.contains(&n) {
            seen.push(n);
            result.push(n.to_string());
        }
    }
    println!("{}", result.join(" "));
}
